client/data/emails/use-get-email-accounts-query.ts

```diff
@@ -1,11 +1,24 @@
 import { useQuery } from 'react-query';
 import wpcom from 'calypso/lib/wp';
+import type { UseQueryOptions } from 'react-query';
 
-type UseQueryParams = Parameters< typeof useQuery >;
+export type Mailbox = {
+	domain: string;
+	mailbox: string;
+};
+
+export type EmailAccount = {
+	account_type: string;
+	emails: Mailbox[];
+};
+
+export type UseGetEmailAccountsQueryData = EmailAccount[];
 
 export const getCacheKey = ( siteId: number | null, domain: string ) => [
-	'emailAccounts',
+	'sites',
 	siteId,
+	'emails',
+	'account',
 	domain,
```

Comment:
I feel that if we are going for predictable cache keys, we should pattern them very closely to the request URLs.
```suggestion
	'accounts',
	domain,
	'mailboxes'
```

Reply:
Sloppiness on my part! The goal was indeed to map 1:1 to the request URL